scripts/common/kubernetes.sh

```diff
@@ -252,18 +252,30 @@ function kubernetes_has_remotes() {
     return 1
 }
 
+# Fetch the load balancer endpoint from the cluster.
+function existing_kubernetes_api_address() {
+    kubectl get cm -n kube-system kurl-config -o jsonpath='{ .data.kubernetes_api_address }'
+}
+
+# During the upgrade user might change the load balancer endpoint or want to use EKCO internal load balancer. So, we
+# to be checking the api endpoint status on the existing api server endpoint as the new endpoint is only available after
+# finishing the upgrade.
 function kubernetes_api_address() {
-    local addr="$LOAD_BALANCER_ADDRESS"
-    local port="$LOAD_BALANCER_PORT"
+    if [ -n "$upgrading_kubernetes" ]; then
```

Comment:
Where is this variable defined? It is scoped as local to the functions upgrade_kubernetes_local_master_patch and upgrade_kubernetes_local_master_minor

Reply:
so i tested this and it works! blows my mind. weird how scoping variables works in bash.

Reply:
@emosbaugh  Sorry didn't get your question exactly. I am defining the variable only in the  upgrade functions and during the normal installation the if condition will result false as the variable is not defined so the value is null. So, continues with the build.